### PR TITLE
word count

### PR DESCRIPTION
**Pull request by nick1123**

### CODE REVIEW

File: 2017_01_04_word_count/wc.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+fs = require('fs')
+fs.readFile(fileName, 'utf8', function (err, data) {
+    if (err) {
+        return console.log(err);
+    }
+
+    console.log(lineCount(data) + ' ' + wordCount(data) + ' ' + totalChars(data) + ' ' + fileName);
+});
+
+function lineCount(data) {
```
Comment (claydiffrient) [inline 2017_01_04_word_count/wc.js:4]: So as a general style thing here, a good force of habit for times it does matter is putting functions before where they are called.  So I'd expect your readFile stuff to be at the bottom.

File: 2017_01_04_word_count/wc.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Run this with:
+// node wc.js quotes.txt
+var fileName = process.argv[2];
+fs = require('fs')
+fs.readFile(fileName, 'utf8', function (err, data) {
+    if (err) {
+        return console.log(err);
+    }
+
+    console.log(lineCount(data) + ' ' + wordCount(data) + ' ' + totalChars(data) + ' ' + fileName);
```
Comment (claydiffrient) [inline 2017_01_04_word_count/wc.js]: So assuming you are on Node 6 or higher, you should be able to use interpolation here instead.

So `${lineCount(data)} ${wordCount(data) ${totalChars(data)} ${fileName}` would work.

**Review by claydiffrient — COMMENTED** [review]
Looks good overall.  Very efficient a couple of minor comments.

File: 2017_01_04_word_count/wc.js (hunk 1 above)
Comment (britton-jb) [reply 2017_01_04_word_count/wc.js:4]: Could you help me out so I can understand this better? Is there an article about what cases would it matter? I hadn't heard that before. My understanding was that function hoisting would ensure no issues here.

Hi Clay, I'm Britton by the way.

Comment (britton-jb) [reply 2017_01_04_word_count/wc.js:4]: Found one after some quick Googling actually. TIL. Thanks Clay.

http://adripofjavascript.com/blog/drips/variable-and-function-hoisting

**Pull request merged**